### PR TITLE
task 1-4 andrei-zhivotkov

### PR DESCRIPTION
**Pull request by Inquisitor1229**
subtask 1-13

### CODE REVIEW

File: task/04-arrays-tasks.js [hunk 1]
```diff
@@ -503,7 +611,23 @@ function distinct(arr) {
  *   }
  */
 function group(array, keySelector, valueSelector) {
-   throw new Error('Not implemented');
+   let key = array.map(keySelector),
+       value = array.map(valueSelector),
+       map = new Map();
+       
+   key.filter((x, i) => {
+      if (!map.has(x)) {
```
Comment (ilya4321) [inline task/04-arrays-tasks.js]: you are using filter as 'for' here. Try with reduce

File: task/04-arrays-tasks.js [hunk 2]
```diff
@@ -519,7 +643,13 @@ function group(array, keySelector, valueSelector) {
  *   ['one','two','three'], x=>x.split('')  =>   ['o','n','e','t','w','o','t','h','r','e','e']
  */
 function selectMany(arr, childrenSelector) {
-    throw new Error('Not implemented');
+   let arr2 = [];
+   arr.map(x => {
+      arr2.push(...childrenSelector(x));
```
Comment (ilya4321) [inline task/04-arrays-tasks.js]: as before, `map` as `for`. That's a hack for the task =) please try reduce

**Review by IlyaMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!
Also, please update name of the PR to follow the style `task1-4 [your nickname in slack]`, because I don't know who you are

**Comment by IlyaMokin** [comment]
Thanks, looks good now, closing the PR

**Pull request closed without merging**